sqlx-macros/src/derives/type.rs

```diff
@@ -59,22 +59,22 @@ fn expand_derive_has_sql_type_transparent(
 
     if attr.transparent {
         let mut generics = generics.clone();
-        generics.params.insert(0, parse_quote!(DB: sqlx::Database));
+        generics.params.insert(0, parse_quote!(DB: ::sqlx::Database));
         generics
             .make_where_clause()
             .predicates
-            .push(parse_quote!(#ty: sqlx::Type<DB>));
+            .push(parse_quote!(#ty: ::sqlx::Type<DB>));
 
         let (impl_generics, _, where_clause) = generics.split_for_impl();
 
         return Ok(quote!(
-            impl #impl_generics sqlx::Type< DB > for #ident #ty_generics #where_clause {
+            impl #impl_generics ::sqlx::Type< DB > for #ident #ty_generics #where_clause {
                 fn type_info() -> DB::TypeInfo {
-                    <#ty as sqlx::Type<DB>>::type_info()
+                    <#ty as ::sqlx::Type<DB>>::type_info()
                 }
 
-                fn compatible(ty: &DB::TypeInfo) -> bool {
-                    <#ty as sqlx::Type<DB>>::compatible(ty)
+                fn compatible(ty: &DB::TypeInfo) -> ::std::bool {
```

Comment:
This should be `::std::primitive::bool`, right?

Reply:
Same for `str` in a few places.

Reply:
> This should be `::std::primitive::bool`, right?

yes, good catch

> Same for `str` in a few places.

I only found one typo I did (`str::str`). If you find anything else I missed please let me know

Reply:
Right, I only found two bool's and one str. I'm sure you found them all :)